token/core/cmd/pp/gen.go

```diff
@@ -173,3 +185,46 @@ func genChaincodePackage(raw []byte) error {
 
 	return nil
 }
+
+type PP interface {
+	AddAuditor(raw view.Identity)
+	AddIssuer(raw view.Identity)
+}
+
+func SetupIssuersAndAuditors(pp PP) error {
+	// Auditors
+	for _, auditor := range Auditors {
+		// Build an MSP Identity
+		entries := strings.Split(auditor, ":")
+		if len(entries) != 2 {
+			return errors.Errorf("invalid auditor [%s]", auditor)
+		}
+		provider, err := x509.NewProvider(entries[0], entries[1], nil)
+		if err != nil {
+			return errors.WithMessagef(err, "failed to create x509 provider for auditor [%s]", auditor)
+		}
+		id, _, err := provider.Identity(nil)
+		if err != nil {
+			return errors.WithMessagef(err, "failed to get auditor identity [%s]", auditor)
+		}
+		pp.AddAuditor(id)
+	}
+	// Issuers
+	for _, issuer := range Issuers {
+		// Build an MSP Identity
+		entries := strings.Split(issuer, ":")
+		if len(entries) != 2 {
+			return errors.Errorf("invalid issuer [%s]", issuer)
+		}
+		provider, err := x509.NewProvider(entries[0], entries[1], nil)
+		if err != nil {
+			return errors.WithMessagef(err, "failed to create x509 provider for issuer [%s]", issuer)
+		}
+		id, _, err := provider.Identity(nil)
+		if err != nil {
+			return errors.WithMessagef(err, "failed to get issuer identity [%s]", issuer)
+		}
+		pp.AddIssuer(id)
+	}
```

Comment:
I guess this can be refactored in a common function. For example:
```go
func getIdentity(s string) (ID, err) {
                entries := strings.Split(s, ":")
		if len(entries) != 2 {
			return nil, errors.Errorf("invalid input [%s]", s)
		}
		provider, err := x509.NewProvider(entries[0], entries[1], nil)
		if err != nil {
			return nil, errors.WithMessagef(err, "failed to create x509 provider for [%s]", s)
		}
		id, _, err := provider.Identity(nil)
		if err != nil {
			return nil, errors.WithMessagef(err, "failed to get identity [%s]", s)
		}
}
```

```go
for _, auditor := range auditors {
  id, err := getIdentity(auditor)
  if err != nil {
    return errors.Wrap(err, "Cannot get auditor identity")
  }
  pp.AddAuditor(id)
}

for _, issuer := range issuers {
....
}
```